akka-http-core/src/main/scala/akka/http/scaladsl/model/Uri.scala

```diff
@@ -875,9 +880,9 @@ object UriRendering {
   def renderAuthority[R <: Rendering](r: R, authority: Authority, path: Path, scheme: String, charset: Charset): r.type =
     if (authority.nonEmpty) {
       import authority._
-      if (!userinfo.isEmpty) encode(r, userinfo, charset, `userinfo-char`) ~~ '@'
+      if (!userinfo.isEmpty) encode(r, userinfo, charset, `userinfo-char`, detectAlreadyEscapedChars = true) ~~ '@'
```

Comment:
This will not work as it introduces ambiguities. We cannot just guess if something accidentally introduced encodings or if the value was meant that way.

Reply:
Yeah, that's the root of my worry here.